### PR TITLE
🚧 themeables/core: scaffolding of theme override

### PR DESCRIPTION
**Pull request by iJigg**
### Todo

- [ ] Decide if this the right approach
- [ ] Type it out 
- [ ] Ship it & never look back

### CODE REVIEW

File: packages/themeables/core/src/index.ts [hunk 1]
```diff
@@ -3,19 +3,37 @@ import type { FC } from 'react'
...
-      (props) => ({
-        ...useContext(ThemePiece)[name](props),
-        ...props,
-      })
+      (props) => {
+        const themeProps = useContext(ThemePiece)[name](props)
+        let overrides = {}
+
+        if (overrideTheme
+          && useContext(overrideTheme)
```
[the remark this answers is not in the document]
Comment (irminakacz) [reply packages/themeables/core/src/index.ts:24]: The full line from documentation says: `A component calling useContext will always re-render *when the context value changes*.` and here we're calling with the same value, so it shouldn't re-render, right?

File: packages/themeables/core/meta.tsx [hunk 2]
```diff
@@ -25,7 +25,9 @@ const defaultTheme: TThemeables<TThemeableBackground, Mappings> = {
   }),
 }
 
-const { ThemePiece, createThemeable } = setupTheme<TThemeableBackground, Mappings>(defaultTheme)
+const OverrideContext: React.Context<TOverrideables<TThemeableBackground, Mappings>> = createContext({})
```
[the remark this answers is not in the document]
Comment (irminakacz) [reply packages/themeables/core/meta.tsx:28]: I'm not sure I follow what kind of helper are you thinking about. Could you elaborate?

File: packages/themeables/core/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (irminakacz) [reply packages/themeables/core/src/index.ts:24]: We can also just test it and count the re-renders to put our minds at ease :)

File: packages/themeables/core/meta.tsx [hunk 3]
```diff
@@ -39,34 +41,45 @@ const newTheme: TThemeables<TThemeableBackground, Mappings> = {
...
-    {(
+const overrideTheme: TOverrideables<TThemeableBackground, Mappings> = {
+  [SYMBOL_DEMO]: ({ status }) => ({
+    color: status === 'error' ? [144, 24, 56, 1] : [108, 105, 86, 1],
+  }),
+}
+
+export const Component = ({ status, hasTheme, withOverride }: TDemoComponent) => {
+  return (
+    <OverrideContext.Provider value={withOverride ? overrideTheme : {}}>
```
Comment (DonGideon) [inline packages/themeables/core/meta.tsx:54]: shouldnt that one be on line 62 and not wrapping it all?

File: packages/themeables/core/src/index.ts [hunk 4]
```diff
@@ -3,19 +3,37 @@ import type { FC } from 'react'
...
   const createThemeable = <P extends ThemeType>(name: K, Target: FC<any>) => {
     const Themeable = component(
       startWithType<Partial<P> & ComponentMappings[K]>(),
-      (props) => ({
-        ...useContext(ThemePiece)[name](props),
-        ...props,
-      })
+      (props) => {
+        const themeProps = useContext(ThemePiece)[name](props)
+        const overrideCtx = useContext(overrideTheme)
```
Comment (DonGideon) [inline packages/themeables/core/src/index.ts:22]: can we use the full name? "overrideContext" ? 
for me its easier to read, wdyt?

**Comment by DonGideon** [comment]
Great and must needed PR 🔥

File: packages/themeables/core/meta.tsx (hunk 3 above)
Comment (iJigg) [reply packages/themeables/core/meta.tsx:54]: I'm not sure why this matters, there is no right/wrong here. Only requirement is that any themeable is wrapped with it. 

But for the sake of removing any confusion i can move it

Comment (DonGideon) [reply packages/themeables/core/meta.tsx:54]: I just think it might make it a bit clearer, thanks for understanding

**Pull request merged**